test/drlg_l1_test.cpp

```diff
@@ -19,3 +26,134 @@ TEST(Drlg_l1, DRLG_Init_Globals)
 	DRLG_Init_Globals();
 	EXPECT_EQ(dLight[0][0], 0);
 }
+
+void TestCreateL5Dungeon(bool hellfire, int level, int seed, lvl_entry entry)
+{
+	pMegaTiles = std::make_unique<MegaTile[]>(1648);
+
+	MyPlayer->pOriginalCathedral = !hellfire;
+
+	currlevel = level;
+	leveltype = DTYPE_CATHEDRAL;
+
+	CreateL5Dungeon(seed, entry);
+
+	std::string path = paths::BasePath();
+
+	paths::SetPrefPath(path);
+	std::string dunPath;
+	if (hellfire)
+		dunPath = fmt::format("../test/fixtures/hellfire/{}-{}.dun", level, seed);
+	else
+		dunPath = fmt::format("../test/fixtures/diablo/{}-{}.dun", level, seed);
+	auto dunData = LoadFileInMem<uint16_t>(dunPath.c_str());
+	EXPECT_EQ(DMAXX, dunData[0]);
+	EXPECT_EQ(DMAXY, dunData[1]);
+
+	const uint16_t *tileLayer = &dunData[2];
+
+	for (int y = 0; y < DMAXY; y++) {
+		for (int x = 0; x < DMAXX; x++) {
+			auto tileId = static_cast<uint8_t>(SDL_SwapLE16(*tileLayer));
+			tileLayer++;
+			EXPECT_EQ(dungeon[x][y], tileId);
+		}
+	}
+
+	const uint16_t *transparentLayer = &dunData[2 + DMAXX * DMAXY * 13];
+
+	for (int y = 16; y < 16 + DMAXY * 2; y++) {
+		for (int x = 16; x < 16 + DMAXX * 2; x++) {
+			auto sectorId = static_cast<uint8_t>(SDL_SwapLE16(*transparentLayer));
+			transparentLayer++;
+			EXPECT_EQ(dTransVal[x][y], sectorId);

```

Comment:
This is going to generate a lot of output if the level diverges, this should probably be ASSERT_EQ as well.
```suggestion
			ASSERT_EQ(dTransVal[x][y], sectorId) << "Room/region indexes don't match";
```

Reply:
Also I don't think diverging here is necessarily a problem, as long as the rooms are the same shape they can be given different region IDs... but handling that requires a bit more care.